src/newgrf_commons.h

```diff
@@ -109,9 +109,9 @@ static const uint TLR_MAX_VAR10 = 7; ///< Maximum value for var 10.
  * In contrast to #DrawTileSprites this struct is for allocated
  * layouts on the heap. It allocates data and frees them on destruction.
  */
-struct NewGRFSpriteLayout : ZeroedMemoryAllocator, DrawTileSprites {
-	std::vector<DrawTileSeqStruct> seq;
-	std::vector<TileLayoutRegisters> registers;
+struct NewGRFSpriteLayout : DrawTileSprites {
```

Comment:
We probably need to make sure `DrawTileSprites` and `PalSpriteID` are properly initialised?